langchain4j-oracle/src/main/java/dev/langchain4j/store/embedding/oracle/OracleEmbeddingStore.java

```diff
@@ -1,542 +1,592 @@
-        this.table = ensureNotBlank(table, "table");
-        this.accuracy = getOrDefault(accuracy, DEFAULT_ACCURACY);
-        this.distanceType = getOrDefault(distanceType, DEFAULT_DISTANCE_TYPE);
-        this.indexType = getOrDefault(indexType, DEFAULT_INDEX_TYPE);
-        this.normalizeVectors = getOrDefault(normalizeVectors, false);
-
-        useIndex = getOrDefault(useIndex, false);
-        createTable = getOrDefault(createTable, true);
-        dropTableFirst = getOrDefault(dropTableFirst, false);
-        dimension = getOrDefault(dimension, DEFAULT_DIMENSIONS);
-
-        initTable(dropTableFirst, createTable, useIndex, dimension);
-    }
+    private OracleEmbeddingStore(Builder builder) {
+        this.dataSource = ensureNotNull(builder.dataSource, "dataSource");
+        this.tableName = ensureNotNull(builder.tableName, "tableName");
 
+        createSchema(dataSource, tableName);
+    }
 
     /**
-     * Adds a given embedding to the store.
+     * <p>
+     * Creates database tables, indexes, and any other schema objects needed to store embeddings. The schema objectsif it does not exist already.
+     * </p><p>
+     * The table uses a VARCHAR(36) column as a primary key. This data type is chosen for consistency with other
+     * embedding store implementations which accept {@link UUID#toString()} as an id.
+     * </p><p>
+     * Embeddings are stored as VECTOR having any length of FLOAT32 dimensions. The FLOAT32 type can store any number
+     * represented in the <code>float[]</code> returned by {@link Embedding#vector()}. A NOT NULL constraint conforms
+     * with the behavior other embedding store implementations which do not accept NULL embeddings for "add" operations.
+     * </p><p>
+     * The text is stored as CLOB, allowing text of any length to be stored (versus VARCHAR which is limited to 32k
+     * characters). A string returned by {@link TextSegment#text()} can be up to 2G characters.
+     * </p><p>
+     * The metadata is stored as JSON. The unstructured JSON type can store the unstructured metadata returned by
+     * {@link TextSegment#metadata()}.
+     * </p><p>
+     * A vector index is created on the embedding column to speed up similarity search queries. The vector
+     * index uses a cosine distance, which is the same metric used by the {@link #search(EmbeddingSearchRequest)}
+     * method.
+     * </p><p>
+     * The vector index type is an inverted flat file (IVF), rather than hierarchical navigable small world (HNSW),
+     * because DML operations are not possible after creating an HNSW index. Methods like {@link #add(Embedding)}
+     * require the use of DML operations.
+     * </p><p>
+     * There are many parameters which can tune the index, but none are set for now. Later work may tune the index for
+     * operations of a an embedding store.
+     * tuned operations values have these parameters set to any non-default values that would tuned for the specific
```

Comment:
Missing capital letter in the beginning of the sentence: "**t**uned operations values have these ..."

Reply:
Fixed in latest commit. Thank you.